### PR TITLE
Chore/code review

### PR DESCRIPTION
**Pull request by 9jaswag**
this is the current state of my app. Feedback will be implemented as they come.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,8 @@
...
-    "presets":[
-        "latest"
-    ]
+  "presets": ["react", "es2015"],
+  "env": {
+    "development": {
+      "presets": ["react-hmre"]
+    }
+  }
 }
```
Comment (BolajiOlajide) [inline .babelrc]: add an end of line here. Also delete the `.DS_Store` above and add it to your `.gitignore` file.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,5 @@
 # dependencies
-/node_modules
\ No newline at end of file
+/node_modules
+
+.DS_Store
+.env
```
Comment (BolajiOlajide) [inline .gitignore]: ### Nitpick

Add a new line here.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+services:
+  - postgresql
+before_script:
+  - createuser chuks -s -U postgres
+  - createdb postit-test --username=chuks
+after_success: 
+  - npm run coverage
+  - npm run coveralls
+notifications:
+  - webhooks: https://coveralls.io/webhook?repo_token=yEvDWmpY7RvxsiiRUxGYcb85eNoYYwTac
```
Comment (BolajiOlajide) [inline .travis.yml]: ### Nitpick

Check your files and make sure you add an extra line to everything so you don't keep seeing this red warning.

### Critical
You just exposed your `repo-token`, you should look into saving in an environment variable instead and not expose it.

File: Procfile [hunk 4]
```diff
@@ -0,0 +1 @@
+npm start
```
Comment (BolajiOlajide) [inline Procfile]: ### Nitpick

Add a new line here

File: client/__test__/actions/readbyAction.test.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import expect from 'expect';
+import updateReadBy from '../../actions/readbyAction';
+
+describe('Message Readby Action', () => {
+  it('should return a function', () => {
```
Comment (andela-aanchirinah) [inline client/__test__/actions/readbyAction.test.jsx]: I like the correlation of your test descriptive messages but can you be more specific to say what the function that is returning actually does, for example, _Message Ready Action should dispatch Message Ready action type with payload_

Comment (andela-aanchirinah) [inline client/__test__/actions/readbyAction.test.jsx]: I like the correlation of your test descriptive messages but can you be more specific to say what the function that is returning actually does, for example, _Message Ready Action should dispatch Message Ready action type with payload_

File: client/__test__/components/PostMessageForm.test.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+describe('Post message form component', () => {
+  const props = {
+    postMessage: jest.fn(() => Promise.resolve())
+  }
+  it('should render without crashing', () => {
+    const component = shallow(<PostMessageForm {...props}/>);
+    expect(component.node.type).toEqual('div');
+  });
+  it('should contain the method onChange', () => {
```
Comment (andela-aanchirinah) [inline client/__test__/components/PostMessageForm.test.jsx]: From the assertion I see you do , you are testing the behaviour of the onChange method too, so I don't think your description should be _should contain onChange method_

File: client/actions/getGroups.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+export const setUserGroups = (groups) => {
+  return {
+    type: GET_USER_GROUPS,
+    groups
+  };
+};
+
+/**
+ * @return {promise} returns an array of groups
```
Comment (andela-aanchirinah) [inline client/actions/getGroups.js]: Improper JSDoc format , it does not contain name of the function.

File: client/actions/getMessages.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Action to get the messages belonging to a group
+ */
+
+import axios from 'axios';
+
+/**
+ * @return {promise} returns an array of messages
```
Comment (andela-aanchirinah) [inline client/actions/getMessages.js]: Improper JSDoc format  it should contain function name and the return statement should be on the last line

File: client/components/App.jsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import HomePage from './home/HomePage';
+import DashboardPage from './dashboard/DashboardPage'
+import GroupPage from './group/GroupPage';
+import SearchPage from './search/Searchpage';
+import MessagePage from './message/MessagePage';
+import ResetPassword from './resetPassword/ResetPassword';
+import requireAuth from '../utilities/requireAuth';
+
+/**
+ * App component class
```
Comment (andela-aanchirinah) [inline client/components/App.jsx]: Add more description

File: client/components/addUser/AddUserForm.jsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
+/**
+ * Component for form that adds a new user to a group
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { findUser } from '../../actions/addUserAction';
+import addUser from '../../actions/addUserAction';
+
+export class AddUserForm extends Component {
```
Comment (andela-aanchirinah) [inline client/components/addUser/AddUserForm.jsx]: No JSDoc on this file

File: client/components/message/MessagePage.jsx [hunk 11]
```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import classnames from 'classnames';
+import Sidebar from '../sidebar/Sidebar';
+
+/**
+ * Message component
+ * for displayng individual messages
+ */
+export class MessagePage extends Component {
```
Comment (andela-aanchirinah) [inline client/components/message/MessagePage.jsx]: No JSDoc on this file

File: client/components/postMessage/PostMessageForm.jsx [hunk 12]
```diff
@@ -0,0 +1,94 @@
+/**
+ * Component for form that posts a message to a group
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import postMessage from '../../actions/postMessageAction';
+
+export class PostMessageForm extends Component {
```
Comment (andela-aanchirinah) [inline client/components/postMessage/PostMessageForm.jsx]: No JSDoc on this file

File: client/index.jsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import jwt from 'jsonwebtoken';
+import setAuthToken from './utilities/setAuthToken';
+import rootReducer from './rootReducer';
+import App from './components/App';
+import { setCurrentUser } from './actions/signinAction';
+import setGroupId from './actions/groupIdAction';
+import passMessage from './actions/passMessageAction';
+import './styles/main.scss';
+
+const store = createStore(
```
Comment (andela-aanchirinah) [inline client/index.jsx]: Do this outside of this file

File: client/reducers/auth.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
+/**
+ * reducer to add flash message to store
+ */
+
+import isEmpty from 'lodash/isEmpty';
+import { SET_CURRENT_USER } from '../actions/types';
+
+const initialState = {
```
Comment (andela-aanchirinah) [inline client/reducers/auth.js]: Do this also outside with the other initialStates you are using in other files

File: client/utilities/requireAuth.jsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Component to protect auth required pages
+ */
+
+import React, { Component } from 'react';
```
Comment (andela-aanchirinah) [inline client/utilities/requireAuth.jsx]: No JSDoc on this file

File: package.json [hunk 16]
```diff
@@ -4,11 +4,17 @@
   "description": "PostIt is a simple application that allows friends and colleagues create groups for notifications.",
   "main": "index.js",
   "scripts": {
-    "start:dev": "nodemon ./server/bin/www",
-    "test": "mocha --compilers js:babel-register",
+    "start:dev": "nodemon --exec babel-node ./server/bin/www",
+    "start": "babel-node ./server/bin/www",
```
Comment (andela-aanchirinah) [inline package.json]: DO NOT USE _babel-node_ IN PRODUCTION

File: package.json [hunk 17]
```diff
@@ -4,11 +4,17 @@
...
+    "test": "NODE_ENV=test sequelize db:migrate && NODE_ENV=test mocha --compilers js:babel-core/register server/test --timeout 50000",
     "lint": "esw server --color",
     "build-css": "node-sass --include-path scss ./template/styles/main.scss ./template/styles/style.css",
-    "watch-css": "nodemon -e scss -x \"npm run build-css\""
+    "watch-css": "nodemon -e scss -x \"npm run build-css\"",
+    "coverage": "nyc report --reporter=text-lcov | coveralls",
```
Comment (andela-aanchirinah) [inline package.json]: the coverage and coveralls scripts , keep one of them not both

**Review by aanchirinah — COMMENTED** [review]
So everywhere you see _No JSDoc found_ means you have to put JSDoc on every class and function in that file and also the reviews apply to so many files that I might not have commented on but all the same excellent work you have done bro, keep it up.  

PS: Improve your code climate GPA to atleast 3.0

File: server/bin/www [hunk 18]
```diff
@@ -2,8 +2,12 @@
 const http = require('http');
 const app = require('../app'); // The express app we just created
 
-const port = parseInt(process.env.PORT, 10) || 8000;
+const port = parseInt(process.env.PORT, 10) || 9000;
```
Comment (kingisaac95) [inline server/bin/www]: I see you're using an environment variable here. You may want to include an env.sample file to describe how the file should look.

**Pull request merged**